src/assignment-logger.ts

```diff
@@ -28,7 +29,7 @@ export interface IAssignmentEvent {
   /**
    * The assigned variation
    */
-  variation: string;
+  variation: string | null;
```

Comment:
When no allocation matches (or when the flag is disabled) and we return `noneResult`. Is the question that you don't expect such occurrenced to be logged?

Reply:
Yes this is to enable logging the lack of assignment, which seems valuable especially because you probably want to figure out why assignments are not working

Reply:
This makes sense if its a change we want. Note that currently, when no assignment is given, we do not log the null assignment.